File: reactnative/reactnative-todo/ios/reactnativetodo/Configuration.swift
/*
 * Copyright (c) 2024 Ping Identity. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

/**
 A struct that holds the configuration constants for the authentication journey.
 */
public struct Configuration {
    /// The main authentication journey name.
    public let mainAuthenticationJourney: String
    /// The URL of the authentication server.
    public let amURL: String
    /// The name of the cookie used for authentication.
    public let cookieName: String
    /// The realm used for authentication.
    public let realm: String
    /// The OAuth client ID.
    public let oauthClientId: String
    /// The OAuth redirect URI.
    public let oauthRedirectURI: String
    /// The OAuth scopes.
    public let oauthScopes: String
    /// The discovery endpoint for OAuth configuration.
    public let discoveryEndpoint: String
    /// The registration service ID
    public let registrationServiceId: String

    /// Initializer to provide external configuration.
    public init(
        mainAuthenticationJourney: String = "[YOUR_MAIN_AUTHENTICATION_JOURNEY_NAME]",
        amURL: String = "[YOUR_AM_URL]",
        cookieName: String = "[COOKIE NAME]",
        realm: String = "[REALM NAME]",
        oauthClientId: String = "[OAUTH_CLIENT_ID]",
        oauthRedirectURI: String = "[OAUTH_REDIRECT_URI]",
        oauthScopes: String = "[OAUTH_SCOPES]",
        discoveryEndpoint: String = "[DISCOVERY_ENDPOINT]",
        registrationServiceId: String = "Registration"
    ) {
        self.mainAuthenticationJourney = mainAuthenticationJourney
        self.amURL = amURL
        self.cookieName = cookieName
        self.realm = realm
        self.oauthClientId = oauthClientId
        self.oauthRedirectURI = oauthRedirectURI
        self.oauthScopes = oauthScopes
        self.discoveryEndpoint = discoveryEndpoint
        self.registrationServiceId = registrationServiceId
    }
}

File: reactnative/reactnative-todo/android/app/src/main/java/com/reactnativetodo/Configuration.kt
/*
 * Copyright (c) 2024 Ping Identity. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

package com.reactnativetodo


/**
 * Configuration class for the ForgeRock SDK.
 */
data class Configuration(
    /** The main authentication journey name.  */
    val mainAuthenticationJourney: String = "jey-webAuthn",

    /** The URL of the authentication server.  */
    val amURL: String = "https://openam-sdks.forgeblocks.com/am",

    /** The name of the cookie used for authentication.  */
    val cookieName: String = "5421aeddf91aa20",

    /** The realm used for authentication.  */
    val realm: String = "alpha",

    /** The OAuth client ID.  */
    val oauthClientId: String = "AndroidTest",

    /** The OAuth redirect URI.  */
    val oauthRedirectURI: String = "org.forgerock.demo:/oauth2redirect",

    /** The OAuth scopes.  */
    val oauthScopes: String = "openid profile email address phone",

    /** The discovery endpoint for OAuth configuration.  */
    val discoveryEndpoint: String = "[DISCOVERY_ENDPOINT]",

    val registrationServiceName: String = "Registration",
)
